ApiResponse implementation (#12)
answear/luigis-box-bundle: README.md

---
In all request you can catch some exceptions:
* `BadRequestException` - bad request,

> * `ToManyItemsException` - make request with fewer items,

Reviewer: `TooMany` - is this typo just here or in the exception as well?

Author: Everywhere :+1:

* `MalformedResponseException` - something went wrong with Luigi's Box api response,
* `ToManyRequestsException` - delay request rate,
* `ServiceUnavailableException`

Consider catching them separately:
```php

use Answear\LuigisBoxBundle\Exception\BadRequestException;
use Answear\LuigisBoxBundle\Exception\ToManyItemsException;
use Answear\LuigisBoxBundle\Exception\MalformedResponseException;
use Answear\LuigisBoxBundle\Exception\ToManyRequestsException;
use Answear\LuigisBoxBundle\Exception\ServiceUnavailableException;

try {
    // ... request
} catch (BadRequestException $e){
    //bad request
    $request = $e->getRequest();
    $response = $e->getResponse();
} catch (ToManyItemsException $e){
    //items limit reached
    $limit = $e->getLimit();
} catch (MalformedResponseException $e){
    //bad response 
    $response = $e->getResponse();
} catch (ToManyRequestsException $e){
    //repeat request after $retryAfter seconds
    $retryAfter = $e->getRetryAfterSeconds();
} catch (ServiceUnavailableException $e){
    //delay request
}

```

Response
------------

`\Answear\LuigisBoxBundle\Response\ApiResponse`:

> * (bool) `$success` - `true` if all documents will pass successfully,

Reviewer: ```suggestion
* (bool) `$success` - `true` if all documents will be passed successfully,
```

* (int) `$okCount` - number of successfully passed documents,
* (int) `$errorsCount` - number of failed documents,
* (array) `$errors` - array of `\Answear\LuigisBoxBundle\Response\ApiResponseError` objects,
* (array) `$rawResponse` - decoded response from api.

`ApiResponseError`:
* (string) `$url` - url of document
* (string) `$type` - type of error (ex. `malformed_input`)
* (string) `$reason` - failure text (ex. `incorrect object format`)
* (array|null) `$causedBy` - specific reason of error (ex. `["url": ["is missing"]]`)

Note!

> If some document will fail, `ApiResponse::$success` will be `false`. Check `$okCount` if you want to know how many documents were been updated and `$errors` to check exactly which documents fails.

Reviewer: ```suggestion
`ApiResponse::$success` will be set to `false` if any of passed documents fails. Check `$okCount` if you want to know how many documents were updated and `$errors` to check exactly which documents failed.
```
